### ISSUE TITLE
jupyter lab build error Invalid regular expression 

### ISSUE THREAD

**Issue by langchao209** [jupyterlab/jupyterlab#14875]
[LabBuildApp] Building in /home/sdev/env/share/jupyter/lab
[LabBuildApp] Node v18.16.0

[LabBuildApp] Yarn configuration loaded.
[LabBuildApp] Building jupyterlab assets (production, minimized)
[LabBuildApp] > node /home/sdev/env/lib/python3.11/site-packages/jupyterlab/staging/yarn.js install
[LabBuildApp] ➤ YN0000: ┌ Resolution step
➤ YN0002: │ @jupyterlab/cell-toolbar-extension@npm:4.0.3 doesn't provide react (p8e52f), requested by @jupyterlab/settingregistry
➤ YN0002: │ @jupyterlab/console-extension@npm:4.0.3 doesn't provide react (paf9bc), requested by @jupyterlab/settingregistry
➤ YN0002: │ @jupyterlab/csvviewer-extension@npm:4.0.3 doesn't provide react (p60a45), requested by @jupyterlab/settingregistry
➤ YN0002: │ @jupyterlab/debugger-extension@npm:4.0.3 doesn't provide react (p6058c), requested by @jupyterlab/settingregistry
➤ YN0002: │ @jupyterlab/documentsearch-extension@npm:4.0.3 doesn't provide react (pef9a0), requested by @jupyterlab/settingregistry
➤ YN0002: │ @jupyterlab/extensionmanager-extension@npm:4.0.3 doesn't provide react (p7e936), requested by @jupyterlab/settingregistry
➤ YN0060: │ @jupyterlab/extensionmanager@npm:4.0.3 provides react (p334e5) with version 18.2.0, which doesn't satisfy what react-paginate requests
➤ YN0002: │ @jupyterlab/filebrowser-extension@npm:4.0.3 doesn't provide react (p5a333), requested by @jupyterlab/settingregistry
➤ YN0002: │ @jupyterlab/fileeditor-extension@npm:4.0.3 doesn't provide react (pecdc3), requested by @jupyterlab/settingregistry
➤ YN0002: │ @jupyterlab/htmlviewer-extension@npm:4.0.3 doesn't provide react (p0fb56), requested by @jupyterlab/settingregistry
➤ YN0002: │ @jupyterlab/json-extension@npm:4.0.3 doesn't provide @types/react (pcd18c), requested by react-json-tree
➤ YN0002: │ @jupyterlab/mainmenu-extension@npm:4.0.3 doesn't provide react (pd6648), requested by @jupyterlab/settingregistry
➤ YN0002: │ @jupyterlab/markdownviewer-extension@npm:4.0.3 doesn't provide react (p2c8aa), requested by @jupyterlab/settingregistry
➤ YN0002: │ @jupyterlab/metadataform-extension@npm:4.0.3 doesn't provide react (pd0ed5), requested by @jupyterlab/settingregistry
➤ YN0002: │ @jupyterlab/metadataform@npm:4.0.3 doesn't provide @rjsf/utils (pc34b4), requested by @rjsf/core
➤ YN0002: │ @jupyterlab/metadataform@npm:4.0.3 doesn't provide @rjsf/utils (pae27e), requested by @rjsf/validator-ajv8
➤ YN0002: │ @jupyterlab/services@npm:7.0.3 doesn't provide react (pc77de), requested by @jupyterlab/settingregistry
➤ YN0002: │ @jupyterlab/settingeditor-extension@npm:4.0.3 doesn't provide react (p2013f), requested by @jupyterlab/settingregistry
➤ YN0002: │ @jupyterlab/statusbar-extension@npm:4.0.3 doesn't provide react (pdf838), requested by @jupyterlab/settingregistry
➤ YN0002: │ @jupyterlab/terminal-extension@npm:4.0.3 doesn't provide react (pc1efd), requested by @jupyterlab/settingregistry
➤ YN0002: │ @jupyterlab/toc-extension@npm:6.0.3 doesn't provide react (p4a4d0), requested by @jupyterlab/settingregistry
➤ YN0002: │ @jupyterlab/translation-extension@npm:4.0.3 doesn't provide react (p00467), requested by @jupyterlab/settingregistry
➤ YN0000: │ Some peer dependencies are incorrectly met; run yarn explain peer-requirements <hash> for details, where <hash> is the six-letter p-prefixed code
➤ YN0000: └ Completed in 0s 564ms
➤ YN0000: ┌ Fetch step
➤ YN0000: └ Completed in 0s 480ms
➤ YN0000: ┌ Link step
➤ YN0000: └ Completed in 0s 553ms
➤ YN0000: Done with warnings in 1s 720ms

[LabBuildApp] > node /home/sdev/env/lib/python3.11/site-packages/jupyterlab/staging/yarn.js dlx yarn-berry-deduplicate -s fewerHighest --fail
[LabBuildApp] ➤ YN0000: ┌ Resolution step
➤ YN0000: └ Completed in 3s 282ms
➤ YN0000: ┌ Fetch step
➤ YN0000: └ Completed
➤ YN0000: ┌ Link step
➤ YN0000: └ Completed in 1s 667ms
➤ YN0000: Done in 5s 87ms

No duplicates found, yarn.lock identical

[LabBuildApp] > node /home/sdev/env/lib/python3.11/site-packages/jupyterlab/staging/yarn.js run build:prod:minimize
[LabBuildApp] [webpack-cli] /home/sdev/env/share/jupyter/lab/staging/node_modules/webpack/lib/cli.js:68
				/(\p{Uppercase_Letter}+|\p{Lowercase_Letter}|\d)(\p{Uppercase_Letter}+)/gu,
				^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^

SyntaxError: Invalid regular expression: /(\p{Uppercase_Letter}+|\p{Lowercase_Letter}|\d)(\p{Uppercase_Letter}+)/: Invalid property name
    at internalCompileFunction (node:internal/vm:73:18)
    at wrapSafe (node:internal/modules/cjs/loader:1176:20)
    at Module._compile (node:internal/modules/cjs/loader:1218:27)
    at Module._extensions..js (node:internal/modules/cjs/loader:1308:10)
    at Module.load (node:internal/modules/cjs/loader:1117:32)
    at Module._load (node:internal/modules/cjs/loader:958:12)
    at Module.require (node:internal/modules/cjs/loader:1141:19)
    at require (node:internal/modules/cjs/helpers:110:18)
    at get cli (/home/sdev/env/share/jupyter/lab/staging/node_modules/webpack/lib/index.js:129:10)
    at Function.cli (/home/sdev/env/share/jupyter/lab/staging/node_modules/webpack/lib/util/memoize.js:22:13)

[LabBuildApp] JupyterLab failed to build
[LabBuildApp] ERROR | Build failed.
Troubleshooting: If the build failed due to an out-of-memory error, you
may be able to fix it by disabling the `dev_build` and/or `minimize` options.

If you are building via the `jupyter lab build` command, you can disable
these options like so:

jupyter lab build --dev-build=False --minimize=False

You can also disable these options for all JupyterLab builds by adding these
lines to a Jupyter config file named `jupyter_config.py`:

c.LabBuildApp.minimize = False
c.LabBuildApp.dev_build = False

If you don't already have a `jupyter_config.py` file, you can create one by
adding a blank file of that name to any of the Jupyter config directories.
The config directories can be listed by running:

jupyter --paths

Explanation:

- `dev-build`: This option controls whether a `dev` or a more streamlined
`production` build is used. This option will default to `False` (i.e., the
`production` build) for most users. However, if you have any labextensions
installed from local files, this option will instead default to `True`.
Explicitly setting `dev-build` to `False` will ensure that the `production`
build is used in all circumstances.

- `minimize`: This option controls whether your JS bundle is minified
during the Webpack build, which helps to improve JupyterLab's overall
performance. However, the minifier plugin used by Webpack is very memory
intensive, so turning it off may help the build finish successfully in
low-memory environments.

[LabBuildApp] Traceback (most recent call last):

[LabBuildApp]   File "/home/sdev/env/lib/python3.11/site-packages/jupyterlab/debuglog.py", line 47, in debug_logging
    yield

[LabBuildApp]   File "/home/sdev/env/lib/python3.11/site-packages/jupyterlab/labapp.py", line 206, in start
    raise e

[LabBuildApp]   File "/home/sdev/env/lib/python3.11/site-packages/jupyterlab/labapp.py", line 197, in start
    build(

[LabBuildApp]   File "/home/sdev/env/lib/python3.11/site-packages/jupyterlab/commands.py", line 507, in build
    return handler.build(
           ^^^^^^^^^^^^^^

[LabBuildApp]   File "/home/sdev/env/lib/python3.11/site-packages/jupyterlab/commands.py", line 723, in build
    raise RuntimeError(msg)

[LabBuildApp] RuntimeError: JupyterLab failed to build

[LabBuildApp] Exiting application: JupyterLab

**Comment by langchao209** [comment]
$ jupyter labextension list

JupyterLab v4.0.3
/home/sdev/env/share/jupyter/labextensions
        jupyterlab-execute-time v3.0.0 enabled OK (python, jupyterlab_execute_time)
        jupyterlab_scenes v2.0.0-alpha1 enabled OK (python, jupyterlab_scenes)


Uninstalled core extensions:
    @jupyterlab/mathjax-extension

Build recommended, please run `jupyter lab build`:
    @jupyterlab/mathjax-extension needs to be removed from build

**Comment by langchao209** [comment]
fixed by changing nodejs to 18.12.1

**Comment by sebacea** [comment]
I have a similar error with node 18.15.0

```
[LabBuildApp] Building in /Users/sebacea/anaconda3/share/jupyter/lab
[LabBuildApp] Node v18.15.0

[LabBuildApp] Yarn configuration loaded.
[LabBuildApp] Building jupyterlab assets (production, minimized)
[LabBuildApp] > node /Users/sebacea/anaconda3/lib/python3.11/site-packages/jupyterlab/staging/yarn.js install --non-interactive
[LabBuildApp] yarn install v1.21.1
[1/5] Validating package.json...
[2/5] Resolving packages...
success Already up-to-date.
Done in 0.29s.

[LabBuildApp] > node /Users/sebacea/anaconda3/lib/python3.11/site-packages/jupyterlab/staging/yarn.js yarn-deduplicate -s fewer --fail
[LabBuildApp] yarn run v1.21.1
$ /Users/sebacea/anaconda3/share/jupyter/lab/staging/node_modules/.bin/yarn-deduplicate -s fewer --fail
Done in 0.54s.

[LabBuildApp] > node /Users/sebacea/anaconda3/lib/python3.11/site-packages/jupyterlab/staging/yarn.js run build:prod:minimize
[LabBuildApp] yarn run v1.21.1
$ webpack --config webpack.prod.minimize.config.js
[webpack-cli] /Users/sebacea/anaconda3/share/jupyter/lab/staging/node_modules/webpack/lib/cli.js:67
                                /(\p{Uppercase_Letter}+|\p{Lowercase_Letter}|\d)(\p{Uppercase_Letter}+)/gu,
                                ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^

SyntaxError: Invalid regular expression: /(\p{Uppercase_Letter}+|\p{Lowercase_Letter}|\d)(\p{Uppercase_Letter}+)/: Invalid property name
    at new Script (node:vm:100:7)
    at NativeCompileCache._moduleCompile (/Users/sebacea/anaconda3/share/jupyter/lab/staging/node_modules/v8-compile-cache/v8-compile-c$
    at Module._compile (/Users/sebacea/anaconda3/share/jupyter/lab/staging/node_modules/v8-compile-cache/v8-compile-cache.js:184:36)
    at Module._extensions..js (node:internal/modules/cjs/loader:1308:10)
    at Module.load (node:internal/modules/cjs/loader:1117:32)
    at Module._load (node:internal/modules/cjs/loader:958:12)
    at Module.require (node:internal/modules/cjs/loader:1141:19)
    at require (/Users/sebacea/anaconda3/share/jupyter/lab/staging/node_modules/v8-compile-cache/v8-compile-cache.js:159:20)
    at get cli (/Users/sebacea/anaconda3/share/jupyter/lab/staging/node_modules/webpack/lib/index.js:120:10)
    at Function.cli (/Users/sebacea/anaconda3/share/jupyter/lab/staging/node_modules/webpack/lib/util/memoize.js:22:13)
error Command failed with exit code 2.
info Visit https://yarnpkg.com/en/docs/cli/run for documentation about this command.

[LabBuildApp] JupyterLab failed to build
[LabBuildApp] Traceback (most recent call last):
```
